packages/pubsub/src/connection-pool.js

```diff
@@ -278,6 +305,59 @@ ConnectionPool.prototype.createMessage = function(connectionId, resp) {
 };
 
 /**
+ * Gets the channels connectivity state and emits channel events accordingly.
+ *
+ * @fires CHANNEL_ERROR_EVENT
+ * @fires CHANNEL_READY_EVENT
+ */
+ConnectionPool.prototype.getAndEmitChannelState = function() {
+  var self = this;
+
+  this.isGettingChannelState = true;
+
+  this.getClient(function(err, client) {
+    if (err) {
+      self.isGettingChannelState = false;
+      self.emit(CHANNEL_ERROR_EVENT);
+      self.emit('error', err);
+      return;
+    }
+
+    var READY_STATE = 2;
+
+    var channel = client.getChannel();
+    var connectivityState = channel.getConnectivityState(false);
+
+    if (connectivityState === READY_STATE) {
+      self.isGettingChannelState = false;
+      self.emit(CHANNEL_READY_EVENT);
+      return;
+    }
+
+    var elapsedTimeWithoutConnection = 0;
+    var now = Date.now();
+    var deadline;
+
+    if (self.noConnectionsTime) {
+      elapsedTimeWithoutConnection = now - self.noConnectionsTime;
+    }
+
+    deadline = now + (300000 - elapsedTimeWithoutConnection);
```

Comment:
Can we alias `300000` to a constant so that we know what/why it is that value?

Reply:
Sure